eval/eval_test.go

```diff
@@ -852,3 +853,35 @@ func TestExtension(t *testing.T) {
 		t.Errorf("object has wrong value.got:\n%s\n---want--\n%s", actual.Value, expected)
 	}
 }
+
+func TestNotCachingErrors(t *testing.T) {
+	s := eval.NewState()
+	_, err := eval.EvalString(s, `func x(n) {aa+n};x(3)`, false)
+	if err == nil {
+		t.Fatalf("should have errored out, got nil")
+	}
+	_, err = eval.EvalString(s, `aa=1;x(4)`, false)
+	if err != nil {
+		t.Errorf("should have not errored out after defining aa, got %v", err)
+	}
+	_, err = eval.EvalString(s, `x(3)`, false)
+	if err != nil {
+		t.Errorf("should have not cached the error, got %v", err)
+	}
+}
+
+func TestNaNMapKey(t *testing.T) {
+	s := eval.NewState()
+	_, err := eval.EvalString(s, `nan=0./0.`, false)
+	if err != nil {
+		t.Errorf("should have not errored out just defining NaN, got %v", err)
+	}
+	_, errs, _ := repl.EvalString(`nan=0./0; m={-42.3: "about -42", 42.1:"42.1", nan: "this is NaN"}; println(m)`)
+	if len(errs) != 1 {
+		t.Errorf("should have an error trying to put a nan in map, got %v", errs)
```

Comment:
```suggestion
		t.Errorf("should have an error trying to put a NaN in map, got %v", errs)
```

Unless you have something against uppercase in log (but I don't think so, as other error are using NaN

Or maybe you love naan, and made a typo 🤣😂

![recette-vegetarienne-cheese-naan-3.jpg](https://github.com/user-attachments/assets/62a7ce53-3c30-421b-8fe6-c9fa9e937a1d)

Reply:
I like naan but yes should be NaN here